src/libusermode/userhook.cpp

```diff
@@ -848,20 +848,162 @@ std::optional<HookActions> get_hook_actions(const std::string& str)
     return std::nullopt;
 }
 
+namespace
+{
+std::optional<std::string> try_parse_token(std::stringstream& ss)
+{
+    const char SEPARATOR = ',';
+    std::string result;
+    if (!std::getline(ss, result, SEPARATOR) || result.empty())
+    {
+        return std::nullopt;
+    }
+    return result;
+}
+
+std::string parse_token(std::stringstream& ss)
+{
+    auto maybe_token = try_parse_token(ss);
+    if (!maybe_token)
+    {
+        throw UserHookException{"Expected a token"};
+    }
+    return *maybe_token;
+}
+
+std::unique_ptr<ArgumentPrinter> make_arg_printer(
+    const PrinterConfig& config,
+    const std::string& type,
+    const std::string& name)
+{
+    if (type == "lpstr" || type == "lpcstr" || type == "lpctstr")
+    {
+        return std::make_unique<AsciiPrinter>(config, name);
+    }
+    else if (type == "lpcwstr" || type == "lpwstr" || type == "bstr")
+    {
+        return std::make_unique<WideStringPrinter>(config, name);
+    }
+    else if (type == "punicode_string")
+    {
+        return std::make_unique<UnicodePrinter>(config, name);
+    }
+    else if (type == "pulong")
+    {
+        return std::make_unique<UlongPrinter>(config, name);
+    }
+    else if (type == "lpvoid*")
+    {
+        return std::make_unique<PointerToPointerPrinter>(config, name);
+    }
+    else if (type == "refclsid" || type == "refiid")
+    {
+        return std::make_unique<GuidPrinter>(config, name);
+    }
+    else if (type == "binary16")
+    {
+        return std::make_unique<Binary16StringPrinter>(config, name);
+    }
+
+    return std::make_unique<ArgumentPrinter>(config, name);
+}
+
+
+std::vector<std::unique_ptr<ArgumentPrinter>> parse_arguments(
+            const PrinterConfig& config,
+            std::stringstream& ss)
+{
+    std::vector<std::unique_ptr<ArgumentPrinter>> argument_printers;
+
+    for (size_t arg_idx = 0; ; arg_idx++)
+    {
+        auto maybe_arg = try_parse_token(ss);
+        if (!maybe_arg) break;
+
+        const std::string arg = *maybe_arg;
+        std::string arg_name;
+        std::string arg_type;
+        const auto pos = arg.find_first_of(':');
+
+        if (pos == std::string::npos)
+        {
+            arg_name = std::string("Arg") + std::to_string(arg_idx);
+            arg_type = arg;
+        }
+        else
+        {
+            arg_name = arg.substr(0, pos);
+            arg_type = arg.substr(pos + 1);
+        }
+
+        argument_printers.emplace_back(make_arg_printer(config, arg_type, arg_name));
+    }
+    return argument_printers;
+}
+
+void parse_entry(
+    plugin_target_config_entry_t& entry,
+    std::stringstream& ss,
+    PrinterConfig& config)
```

Comment:
This looks very C-like. Shouldn't it be:
```suggestion
plugin_target_config_entry_t parse_entry(
    const std::string& line,
    const PrinterConfig& config)
```
?

Reply:
true